deploy/olm-catalog/observatorium-operator/0.0.1/core.observatorium.io_observatoria.yaml

```diff
@@ -0,0 +1,803 @@
+---
```

Comment:
hmm, do we really need yet another copy of this file in this repo? we have one already at https://github.com/observatorium/configuration/blob/master/deploy/crds/core.observatorium.io_observatoria.yaml

Reply:
Unfortunately we need, for upgrades - please see [our comment on a previous thread](https://github.com/observatorium/configuration/pull/269#discussion_r425150683)

Reply:
We can avoid this if we say that we don't care about upgrades from previous versions (but then why would you want OLM, right?) or wait for image indexing to be introduced which is AFAIU planned for Openshift 4.6. If we want to ship this before image indexing is introduced and support upgrades, I don't really see how we can easily (without checking out older branches) avoid adding that to the repo, that's what other release operators we checked do in order to have access to older versions' CRDs during the creation of the release container...

Reply:
@squat but if you have a suggestion for how to do it, we'd love to hear - it's not like we like this solution...

Reply:
The first solution I imagine is to use multistage dockerfiles for all of the OLM containers and for the 0.2.0 container to have a stage that copies a file from the 0.1.0 container, ie the manifest from the previous release. Then on the next release, eg 0.3.0, we need to swap out the FROM to point to 0.2.0. This way we always keep the manifests in the containers as build artifacts and they have the benefit of being inherently tagged and generated without needed to be checked into version control.

Reply:
Ah I see, you propose to retrieve the artifacts for older versions from the image of the most recent release container. That could work but I see some possible issues with that:
1. Assuming we want to base the latest release container on the latest version of openshift/origin-operator-registry, it means that we'll need to download two images that are no tiny rather than just one.
2. Assuming we want to use the latest (or a recent) version of the operator-sdk, we would need to provide a way to access it during the generation of downstream release.

I wouldn't suggest to do that unless there is a good reason not to store few more textual files in the repo.

Reply:
Hmm 1 doesn't sound like a problem. Docker will download two images during building, which is a one-time cost in CI, but not at run time, so this can be ignored IMO, this is what build containers are designed for. 
2 also doesn't sound like a limitation to me, but I understand if this is complexity that you don't want to undertake

Reply:
I think that having the older versions easily available and human readable is valuable.
Someone could find more easily in which version a specific field was added to the CRD for example.

Also, for example, see that all the versions are available:

https://github.com/operator-framework/community-operators/tree/master/upstream-community-operators/prometheus

https://github.com/kubevirt/hyperconverged-cluster-operator/tree/master/deploy/olm-catalog/kubevirt-hyperconverged